### ISSUE TITLE
Adding support for FreeText annotations

### ISSUE THREAD

**Issue by irfanajanovic** [py-pdf/fpdf2#949]
Hi there,

I am trying to add a text annotation to a document, and have it display. Instead, I am getting an icon on the pdf that shows the value when I hover over it.

I have got this successfully working with another library (pdf_annotate), but unfortunately when building a standalone executable, that library is giving me issues.

I have attached two images (one is what I am trying to accomplish, the other is what I am getting with fpdf2). I have been going through the documentation, but I can't seem to find what would change this.

If anyone has any insight, that would be greatly appreciated.

Regards,

Irfan
![pdf_annotate](https://github.com/py-pdf/fpdf2/assets/24856466/794ee303-0f24-4259-bbd1-e28145fb222d)
![fpdf2](https://github.com/py-pdf/fpdf2/assets/24856466/253c716e-eb3e-4060-b2a0-56704c32e0df)

**Comment by Lucas-C** [comment]
Hi @irfanajanovic

I'd be happy to try to help you with this annoying situation 🙂
Maybe we could use this opportunity to improve `fpdf2` regarding annotations.

I will need you to provide use a few more informations please:
* what PDF viewer are using ? Adobe acrobat ? A web browser? [Sumatra PDF reader](https://www.sumatrapdfreader.org/free-pdf-reader) ?
* what version of `fpdf2` & Python are you using?
* could you please provide some minimal code source that you are using to generate your PDF file with `fpdf2`? Or you using exactly the same code as some section of our documentation?
* and finally, you mentioned that with another library you were able to get better results: could please provide the resulting PDF file, for comparison?

**Comment by Lucas-C** [comment]
Thank you for those detailed information @irfanajanovic

Those are the annotations dictionaries embedded in those files, extracted & pretty-formatted using `qpdf --qdf`:
* `issue_949_fpdf2.pdf`:
```
<<
  /Contents (test)
  /F 4
  /Rect [425.2 416.69 428.03 413.86]
  /Type /Annot
  /Subtype /Text
  /Border [0 0 0]
>>
```
* `issue_949_pdf_annotate.pdf`:
```
<<
  /Contents (test)
  /F 4
  /Rect [400 420 600 440]
  /Type /Annot
  /Subtype /FreeText
  /AP <<
    /N 38 0 R
  >>
  /BS <<
    /S /S
    /Type /Border
    /W 0
  >>
  /C []
  /CreationDate (D:20231011151757+00'00)
  /DA (0.6705882353 0.2117647059 0.6352941176 rg /PDFANNOTATORFONT1 11 Tf)
  /M (D:20231011151757+00'00)
  /NM (a8e3f17f-1488-456b-80de-1ed8f1abb9f8)
  /P 16 0 R
>>
```
I notice a few things:
* the annotation dictionary inserted by `pdf_annotate` is a `FreeText`, whereas the one inserted by `fpdf2` is a `Text` annotation
Quoting the PDF spec:
> A free text annotation (PDF 1.3) displays text directly on the page. Unlike an ordinary text annotation (see
12.5.6.4, “Text Annotations”), a free text annotation has no open or closed state; instead of being displayed in a
pop-up window, the text shall be always visible.
* the annotation dictionary inserted by `pdf_annotate` contains some extra fields, including:
  + `/DA` is a the _default appearance string that shall be used in formatting the text_
  + `/AP` is the _appearance dictionary specifying how the annotation shall be presented visually on the page_
  + `/BS` is the _border style dictionary specifying the line width and dash pattern that shall be used in drawing the annotation’s border_

We already considered adding support appearance dictionaries to `fpdf2`: https://github.com/py-pdf/fpdf2/issues/561

`FreeText` annotations are currently not supported by `fpdf2`,
but I'm favorable that we add support for them! 🙂
And it shouldn't be too difficult, given that we already insert several kind of `AnnotationDict` in PDF documents.

**Comment by MarekT0v** [comment]
Hello, is this issue open to work on? If yes, can you assign me to this issue?

**Comment by Lucas-C** [comment]
Hi @MarekT0v

I'm waiting for @irfanajanovic first, in order to know if he wants to work on this feature 🙂
If he does not, I'll be happy to assign this issue to you in if you want to contribute!

**Comment by irfanajanovic** [comment]
Hi @Lucas-C  / @MarekT0v,

Please feel free to take on the feature. I don't have time to work on this at the moment.

Thanks,

Irfan

**Comment by Lucas-C** [comment]
Alright, thank you for your answer and for reporting this @irfanajanovic 🙂

@MarekT0v I assigned this issue to you!

**Comment by Lucas-C** [comment]
Hi @MarekT0v!

Are you still working on this?
Or can other `fpdf2` contributors work on this?

**Comment by MarekT0v** [comment]
Hi @Lucas-C 
I am still working on this. I will try to finish soon, I will inform you.

**Comment by Lucas-C** [comment]
>  I am still working on this. I will try to finish soon, I will inform you.

Great!
You rely on our (the maintainers) help if you need! 😊

**Comment by MarekT0v** [comment]
acb4364161a36763daed9112da8dc4c6358c6f7c

**Comment by Lucas-C** [comment]
> progress, [acb4364](https://github.com/py-pdf/fpdf2/commit/acb4364161a36763daed9112da8dc4c6358c6f7c)

Nice!

Would you like to open a Pull Request, maybe in Draft mode, and we will give you some review feedbacks?

**Comment by Lucas-C** [comment]
This has been implemented by @MarekT0v in PR #1039

Some documention regarding this new feature has been added there:
https://py-pdf.github.io/fpdf2/Annotations.html#free-text-annotations

@irfanajanovic: would you like to test this feature and tell us if it fits your need?
It's not released yet, but you can install the latest version from the `master` branch of this repo by running this command:

    pip install git+https://github.com/py-pdf/fpdf2.git@master
